libs/isograph-react/src/core/garbageCollection.ts

```diff
@@ -51,31 +54,54 @@ export function garbageCollectEnvironment(environment: IsographEnvironment) {
     recordReachableIds(environment.store, query, retainedIds);
   }
 
-  for (const dataId in environment.store) {
-    if (!retainedIds.has(dataId)) {
-      delete environment.store[dataId];
+  for (const typeName in environment.store) {
+    const dataById = environment.store[typeName];
+    if (dataById == null) continue;
```

Comment:
should we throw (eventually, just in dev) if `dataById == null` and `retainedTypeIds != null ? retainedTypeIds.length > 0 : false`?